### ISSUE TITLE
Build a Page Collection module

### ISSUE THREAD

**Issue by nengberg** [cofoundry-cms/cofoundry#86]
Okay, this is not an issue but rather a feature request.

We have the scenario where we want to link to other pages in some way. A suggested implementation would be a module for a page collection, like the Custom Entity Collection, where you can select multiple pages from the solution. When you then have a list of pages it would be great if you could query the page in some way to get additional information. What do you think?

Please let me know if you need any clarification :)

**Comment by HeyJoel** [comment]
Certainly we should have a PageCollectionAttribute that works in a similar way to CustomEntityCollection that includes a picker and drag/drop ordering. A page module type based off this would also be a good idea, but in reality the implementator will want to customize the template and the view data so this might be better off as an example module that can be adapted.

I can also see that it would be good to have a module that queries pages based on other attributes such as directory or tags. We have some raw page querying functionality build in already using 'IPageRepository' based on the demands of the page listing in the admin panel but this area needs a bit more thought, which is linked to [issue 12: Custom Entity Partitioning & Taxonomy](https://github.com/cofoundry-cms/cofoundry/issues/12). For now it would be good to at least have a sample for how this existing functionality could be used.

**Comment by nengberg** [comment]
Yeah, that was exactly what I meant. An attribute to be able to alter the property type. Of course the module itself should be up to the consumer to define! :)

**Comment by HeyJoel** [comment]
I've added three new data model attributes:

- PageAttribute
- PageCollectionAttribute
- WebDirectoryAttribute

I've also created a sample application demonstrating how to use them: [Cofoundry.Samples.PageModules](https://github.com/cofoundry-cms/Cofoundry.Samples.PageModules)

Currently these changes are in pre-release so you'll need to get the changes using our [pre-release feed](https://www.myget.org/F/cofoundry/api/v3/index.json). The package version is [0.1.5-page-data-attrib0006](http://myget.org/feed/cofoundry/package/nuget/Cofoundry.Web.Admin/0.1.5-page-data-attrib0006).

Hopefully I can push a release out in the next few days.
